atomicfu/src/commonTest/kotlin/kotlinx/atomicfu/test/UncheckedCastTest.kt

```diff
@@ -0,0 +1,28 @@
+package kotlinx.atomicfu.test
+
+import kotlinx.atomicfu.AtomicRef
+import kotlinx.atomicfu.atomic
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+private val topLevelS = atomic<Any>(listOf("A", "B"))
+
+class UncheckedCastTest {
+    private val s = atomic<Any>("AAA")
+    private val bs = atomic<Any?>(null)
+
+    @Test
+    fun testAtomicValUncheckedCast() {
+        assertEquals((s as AtomicRef<String>).value, "AAA")
```

Comment:
Please, add a test on using it with `inline` function. It works perfectly now, but a test would still be extremely helpful to have for future maintenance. Add something like this:

```
    inline fun <T> AtomicRef<T>.getString(): String =
        (this as AtomicRef<String>).value
```

The test line here:

```
        assertEquals(s.getString(), "AAA")
```